lib/spack/spack/test/environment_modifications.py

```diff
@@ -158,75 +181,155 @@ def test_unset(env):
+                    "C:\\",
+                    "C:\\Program Files",
+                    "C:\\Program Files (x86)",
+                    "C:\\Users",
+                    "C:\\ProgramData",
+                    "C:\\dev\\spack_window",
+                    "C:\\dev\\spack_window\\lib",
+                ]
+            ),
+        ),
+        # Mac Paths
+        (
+            miscellaneous_paths,
+            (
+                miscellaneous_paths
+                if sys.platform == "win32"
+                else [
+                    "/usr/local/Cellar/gcc/5.3.0/lib",
+                    "/usr/local/opt/some-package/lib",
+                    "/usr/opt/lib",
+                    "/opt/some-package/include",
+                    "/opt/some-package/local/..",
+                ]
+            ),
+        ),
+    ],
+)
+def test_filter_system_paths(miscellaneous_system_paths, expected):
     """Tests that the filtering of system paths works as expected."""
-    filtered = filter_system_paths(miscellaneous_paths)
-    expected = [
-        "/usr/local/Cellar/gcc/5.3.0/lib",
-        "/usr/local/opt/some-package/lib",
-        "/usr/opt/lib",
-        "/opt/some-package/include",
-        "/opt/some-package/local/..",
-    ]
+    filtered = filter_system_paths(miscellaneous_system_paths)
     assert filtered == expected
 
 
-# TODO 27021
-@pytest.mark.not_on_windows("Not supported on Windows (yet)")
-def test_set_path(env):
+@pytest.mark.parametrize("name,elements,separator", [("A", ["foo", "bar", "baz"], os.pathsep)])
+def test_set_path(env, name, elements, separator):
     """Tests setting paths in an environment variable."""
 
-    # Check setting paths with the default separator
-    env.set_path("A", ["foo", "bar", "baz"])
+    # Check setting paths with a specific separator
+    env.set_path(name, elements, separator=separator)
     env.apply_modifications()
 
-    assert "foo:bar:baz" == os.environ["A"]
-
-    env.set_path("B", ["foo", "bar", "baz"], separator=";")
-    env.apply_modifications()
+    expected = os.pathsep.join(elements)
+    assert expected == os.environ[name]
 
-    assert "foo;bar;baz" == os.environ["B"]
 
-
-@pytest.mark.not_on_windows("Not supported on Windows (yet)")
-def test_path_manipulation(env):
+@pytest.mark.parametrize(
+    "path_name,elements,expected",
+    [
+        (
+            "PATH_LIST",
+            ["first", "fourth", "last"],
+            make_pathlist(
+                [
+                    ["path", "first"],
+                    ["path", "second"],
+                    ["path", "third"],
+                    ["path", "fourth"],
+                    ["path", "last"],
+                ]
+            ),
+        ),
+        (
+            "EMPTY_PATH_LIST",
+            ["first", "middle", "last"],
+            make_pathlist([["path", "first"], ["path", "middle"], ["path", "last"]]),
+        ),
+        (
+            "NEWLY_CREATED_PATH_LIST",
+            ["first", "middle", "last"],
```

Comment:
(question) could you call make_pathlist on this?

Reply:
No problem

Reply:
Actually the way this is used by the test I think it makes sense to keep as is.